### ISSUE TITLE
Restore fails with NU1004 after running nuget restore with -ForceEvaluate

### ISSUE THREAD

**Issue by madelson** [NuGet/Home#10374]
## Details about Problem

This has some of the same symptoms as https://github.com/NuGet/Home/issues/9332 but may not be the same issue; it happened under different circumstances and this time I was able to assemble a small solution that can reproduce the problem.

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): NuGet.exe

NuGet version (x.x.x.xxx):  5.8.0.6930

dotnet.exe --version (if appropriate): n/a

VS version (if appropriate): 16.8.3

OS version (i.e. win10 v1607 (14393.321)): Windows 10

Worked before? If so, with which NuGet version: 5.3?

## Detailed repro steps so we can see the same problem

1. Unzip attached Repro.zip archive

2. Restore the repro solution with nuget.exe and -ForceEvaluate => should succeed
```
nuget restore -ForceEvaluate Repro.sln
```

3. Restore the repro solution with nuget.exe and -LockedMode => should FAIL!
```
nuget restore -LockedMode Repro.sln
```

### Verbose Logs
```
NU1004: The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.
NU1004: The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.

```

### Sample Solution
[Repro.zip](https://github.com/NuGet/Home/files/5680970/Repro.zip)

**Comment by kartheekp-ms** [comment]
I can reproduce this issue following the repro steps.

**Comment by nkolev92** [comment]
Dup of https://github.com/NuGet/Home/issues/10326. 


Fixed in 5.9.0-preview.2 https://dist.nuget.org/win-x86-commandline/v5.9.0-preview.2/nuget.exe
